### PR TITLE
[test] Update instance types due to low p2.8x availability

### PR DESCRIPTION
**Pull request by arjkesh**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]

*Description:*
Update test logic to skip p2.8xlarge in mainline context for ec2. Additionally, change instance types to p3.8x where necessary.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -415,6 +415,13 @@ def get_dlc_images():
         test_images = json.load(test_env)
     for dlc_test_type, images in test_images.items():
         if dlc_test_type == "sanity":
+            # TODO: Remove if re-enabling p2.8x tests
+            if os.getenv("EC2_GPU_INSTANCE_TYPE") == "p2.8xlarge":
+                non_gpu_images = []
+                for image in images:
+                    if "gpu" not in image:
+                        non_gpu_images.append(image)
+                return " ".join(non_gpu_images)
```
Comment (saimidu) [inline test/test_utils/__init__.py:424]: Why is this necessary? p2.8xlarge is not assigned as an `EC2_GPU_INSTANCE_TYPE` for sanity tests.
There is no default instance type for sanity tests, right?

File: test/test_utils/sagemaker.py [hunk 2]
```diff
@@ -33,10 +33,12 @@ class DLCSageMakerRemoteTestFailure(Exception):
...
 def assign_sagemaker_remote_job_instance_type(image):
-    if "tensorflow" in image:
-        return "ml.p3.8xlarge" if "gpu" in image else "ml.c4.4xlarge"
+    if "gpu" in image:
+        return "ml.p3.8xlarge"
+    elif "tensorflow" in image:
+        return "ml.c4.4xlarge"
     else:
-        return "ml.p2.8xlarge" if "gpu" in image else "ml.c4.8xlarge"
+        return "ml.c4.8xlarge"
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:41]: nit: Why turn this into a regular if-else block instead of the keeping it as the ternary-if operator like before?

Comment (arjkesh) [reply test/test_utils/sagemaker.py:41]: more clear this way in my opinion

File: test/test_utils/__init__.py (hunk 1 above)
Comment (arjkesh) [reply test/test_utils/__init__.py:424]: This is unrelated to sanity tests - we are getting all the images from the sanity section of the test env file to unify w/ changeset logic. This can be refactored to be more clear as we update our changeset logic (i.e. add a section to the env file called all_images)

what this does is remove gpu images from dlc_images if EC2_GPU_INSTANCE_TYPE is p2.8x

File: test/test_utils/sagemaker.py (hunk 2 above)
Comment (saimidu) [reply test/test_utils/sagemaker.py:41]: Sure.

**Review by saimidu — DISMISSED** [review]

**Comment by saimidu** [comment]
@arjkesh What needs to be done here to get this PR to a Review-Ready state?

**Comment by arjkesh** [comment]
> @arjkesh What needs to be done here to get this PR to a Review-Ready state?

I removed skipping all p2.8x, but kept the change in PR context. It should be ready for review now.

**Comment by saimidu** [comment]
After this PR, what still continues to use p2.8xlarge instances?
